src/firstboot/firstboot.c

```diff
@@ -244,7 +246,36 @@ static int should_configure(int dir_fd, const char *filename) {
                 return true; /* missing */
         }
 
-        return arg_force; /* exists, but if --force was given we should still configure the file. */
+        if (arg_force)
+                return true; /* exists, but if --force was given we should still configure the file. */
+
+        if (streq(filename, "passwd")) {
+                _cleanup_fclose_ FILE *passwd = NULL;
+
+                r = xfopenat(dir_fd, filename, "re", O_NOFOLLOW, &passwd);
+                if (r < 0 && r != -ENOENT)
```

Comment:
Shouldn't we return true here if we get ENOENT?

Reply:
We already did a check with `faccessat` earlier, so we can't get ENOENT really. But this was sloppy. I reworked the code to avoid TOCTOU by doing either `faccessat` or `xfopenat` at once.